### PR TITLE
Labs production refactor

### PR DESCRIPTION
**Pull request by berniedurfee**
# Description
Refactoring in prep for production launch

### CODE REVIEW

**Comment by jessmartin** [comment]
This looks awesome @berniedurfee! 😍 I'll definitely be digging into this soon to give it a second set of eyes. 

Without having gone through it line-by-line, it seems like a lot of these are breaking changes that we'll need to push back to the student repo. Is that what you're thinking?

**Comment by berniedurfee** [comment]
Lots of breaking changes, yes. In fact, lots of the functionality is still broken and needs to be updated. I'm going to push more stuff into this PR to get all the functionality working, but with an updated data model and logic.

File: apollo/src/context/__mocks__/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Generate Jest mock for the context module
```
Comment (tinomen) [inline apollo/src/context/__mocks__/index.js:1]: Wow, Could you put a video together of setting up testing for prismatopia. Having mocks sure would help.

File: apollo/src/datasources/GitHubAPI.js [hunk 2]
```diff
@@ -101,13 +123,12 @@ class GitHubAPI extends GraphQLDataSource {
       const res = await this.query(SPARKLINE, {
         variables: {
           owner,
-          name
-        }
+          name,
```
Comment (tinomen) [inline apollo/src/datasources/GitHubAPI.js:126]: Isn't this an ESLint standard? ;)

File: apollo/src/index.js [hunk 3]
```diff
@@ -13,6 +20,15 @@ const dataSources = require('./datasources');
     dataSources, // Adds our CodeClimateAPI class to our server.
     // ↑↑ All dataSources are attached to the context object
     cors: true,
+    formatError: err => {
+      // Don't give the specific errors to the client.
+      console.log('%O', err);
```
Comment (tinomen) [inline apollo/src/index.js:25]: No winston logger?

File: apollo/src/index.js [hunk 4]
```diff
@@ -13,6 +20,15 @@ const dataSources = require('./datasources');
     dataSources, // Adds our CodeClimateAPI class to our server.
     // ↑↑ All dataSources are attached to the context object
     cors: true,
+    formatError: err => {
+      // Don't give the specific errors to the client.
```
Comment (tinomen) [inline apollo/src/index.js:24]: The error is being passed through this callback function. Was there an intent to replace the err with something that doesn't have the specific error?

File: apollo/src/resolvers/Note.js [hunk 5]
```diff
@@ -1,20 +1,28 @@
...
+ * @returns { import('../generated/prisma-client').Person }
+ */
+const author = (parent, _, context) => {
+  return context.prisma.note({ id: parent.id }).author();
 };
 
-const project = ({ id }, args, context) => {
-  const res = context.prisma.note({ id }).project();
-  return res;
+// const attendedBy = (parent, args, context) => {
```
Comment (tinomen) [inline apollo/src/resolvers/Note.js:11]: Any particular reason you committed commented out code? ;)

File: apollo/src/resolvers/Person.js [hunk 6]
```diff
@@ -1,25 +1,25 @@
 // Resolves all relational fields on type Person
 // where the name of the function is an exact match to the field
 
-const role = (parent, args, context) => {
-  const res = context.prisma.person({ id: parent.id }).role();
-  return res;
-};
+// const role = (parent, args, context) => {
```
Comment (tinomen) [inline apollo/src/resolvers/Person.js:4]: My eyes! 🤕 #commented-out-code

File: apollo/src/resolvers/Product.js [hunk 7]
```diff
@@ -1,52 +1,59 @@
-// Resolves all relational fields on type Product
-// where the name of the function is an exact match to the field
-
-const program = (parent, args, context) => {
-  const res = context.prisma.product({ id: parent.id }).program();
-
-  return res;
+// @ts-check
+
+/**
```
Comment (tinomen) [inline apollo/src/resolvers/Product.js:3]: BTW These function docs are awesome. Thank you.

File: apollo/src/resolvers/Product.js [hunk 8]
```diff
@@ -1,52 +1,59 @@
...
-      }
-      const name = ccRepo.data[0].attributes.human_name;
-      const link = ccRepo.data[0].links.self;
-      return { ...ccSnapshot, name, link, GHRepoId };
-    });
-  } catch (error) {
-    throw new Error(error);
-  }
-};
+// const grades = async (parent, args, context) => {
```
Comment (tinomen) [inline apollo/src/resolvers/Product.js:30]: This commented out code is less awesome. 😆

File: apollo/src/resolvers/Project.js [hunk 9]
```diff
@@ -1,56 +1,76 @@
...
 
-const product = (parent, args, context) => {
+/**
+ * @param { import('../context').ApolloContext } context
+ * @returns { import('../generated/prisma-client').ProductPromise }
+ */
+const product = (parent, _, context) => {
   const res = context.prisma.project({ id: parent.id }).product();
 
   return res;
```
Comment (tinomen) [inline apollo/src/resolvers/Project.js:13]: Should we replace all of these with a single line similar to `apollo/src/resolvers/Program.js`?
```suggestion
  return context.prisma.project({ id: parent.id }).product();
```

File: apollo/src/schema.js [hunk 10]
```diff
@@ -94,15 +92,15 @@ const typeDefs = gql`
...
-  type Label {
-    id: ID!
-    createdAt: String!
-    updatedAt: String!
-    name: String!
-    color: String!
-    status: Status!
-    selected: [Project]
-  }
+  # type Label {
```
Comment (tinomen) [inline apollo/src/schema.js:95]: worth keeping?

File: prisma/seed.js [hunk 11]
```diff
@@ -0,0 +1,291 @@
+/* eslint-disable no-await-in-loop */
```
Comment (tinomen) [inline prisma/seed.js:1]: This seed file is brilliant!

**Review by tinomen — CHANGES_REQUESTED** [review]
Wow, some great changes in here. I hadn't known about winston logger and it's great to see the prisma client mock.

**Comment by jessmartin** [comment]
Very Nice

**Pull request merged**